framework/src/modules/poa/types.ts

```diff
@@ -11,3 +11,52 @@
  *
  * Removal or modification of this copyright notice is prohibited.
  */
+
+import { ImmutableMethodContext, MethodContext } from '../../state_machine';
+import { NextValidatorsSetter } from '../../state_machine/types';
+import { ValidatorKeys } from '../pos/types';
+
+export interface RegisterAuthorityParams {
+	name: string;
+	blsKey: Buffer;
+	generatorKey: Buffer;
+	proofOfPossession: Buffer;
+}
+
+export interface ValidatorsMethod {
+	// https://github.com/LiskHQ/lips/blob/main/proposals/lip-0044.md#setvalidatorgeneratorkey
+	setValidatorGeneratorKey(
+		methodContext: MethodContext,
+		validatorAddress: Buffer,
+		generatorKey: Buffer,
+	): Promise<boolean>;
+	// https://github.com/LiskHQ/lips/blob/main/proposals/lip-0044.md#registervalidatorkeys
+	registerValidatorKeys(
+		methodContext: MethodContext,
+		validatorAddress: Buffer,
+		blsKey: Buffer,
```

Comment:
Note that [the LIP](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0044.md#registervalidatorkeys) gives the following order of parameters: `validatorAddress, proofOfPossession, generatorKey, blsKey` which is different from the order that you mention in the interface here. Please make sure that your code is consistent with the SDK (no need to be consistent with the LIP IMO).

Reply:
You are right. However on the validators module it is arranged that way. Maybe we should open a new issue to re-align this function to the LIP

Reply:
I see no problem if the implementation differs from the LIP in this small way, the important bit is that the protocol is exactly the same.